kythe/rust/indexer/src/indexer/analyzers.rs

```diff
@@ -118,14 +118,22 @@ impl<'a> UnitAnalyzer<'a> {
     /// generate the OffsetIndex
     pub fn handle_files(&mut self) -> Result<(), KytheError> {
         // https://kythe.io/docs/schema/#file
-        for source_file in self.unit.get_source_file() {
-            let vname_result = self.get_file_vname(source_file);
+        for required_input in self.unit.get_required_input() {
+            let info_path = required_input.get_info().get_path();
+            let vname_result = self.get_file_vname(info_path);
+
             // Generated files won't have a file vname returned
             if vname_result.is_err() {
                 continue;
             }
             let vname = vname_result.unwrap();
 
+            // Not all required inputs are Rust source files, so ignore if not
+            // a source file
+            if !info_path.ends_with(".rs") {
+                continue;
+            }
+
             // Create the file node fact
             self.emitter.emit_fact(&vname, "/kythe/node/kind", b"file".to_vec())?;
```

Comment:
The indexer should only emit file nodes (among other things) for the files listed in the `source_file` field, not all of `required_input`.